30411/Nagy N. Hanna/Assignment 1.txt

```diff
@@ -0,0 +1,283 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+FILE *f;
+
+typedef struct node
+{
+    int code;
+    struct node* next;
+
+} nodeT;
+
+nodeT* head, *tail;
+
+
+void addFirst(int code)
+{
+    if(head==NULL)
+    {
+        head=(nodeT*)malloc(sizeof(nodeT*));
+        head->code=code;
+        head->next=NULL;
+        tail=head;
+    }
+    else if(tail==NULL)
```

Comment:
we can somehow eliminate this second if branch.
try putting ```tail=head``` in the first if branch as a last statement and remove this whole branch entirely. Does it work? why?